packages/rich-list-core/src/RichListCore.ts

```diff
@@ -0,0 +1,231 @@
+import { ApiClient, BigNumber, blockchain as defid } from '@defichain/jellyfish-api-core'
+import { WhaleApiClient } from '@defichain/whale-api-client'
+import { QueueClient, Queue } from './lib/Queue'
+import { SingleIndexDb, Schema } from './lib/SingleIndexDb'
+import { AddressParser } from './controller/AddressParser'
+import { NetworkName } from '@defichain/jellyfish-network'
+import { AccountAmount } from '@defichain/jellyfish-api-core/src/category/account'
+import { AddressBalance, ActiveAddressAccountAmount } from './types'
+
+const DEFAULT_RICH_LIST_LENGTH = 1000
+
+export class RichListCore {
+  isCatchingUp = false
+  richListLength = DEFAULT_RICH_LIST_LENGTH
+  readonly addressParser: AddressParser
+
+  constructor (
+    private readonly network: NetworkName,
+    private readonly apiClient: ApiClient,
```

Comment:
Hmmm, it's much more scalable to expose a WhaleClient than a RpcClient. RpcClient we need to replicate an Ocean like design for the `rich-list-api` service with a snapshot provider. It might not be easy. Maybe we can test with WhaleApiClient instead?